### PR TITLE
comments for biased_sampler

### PR DESCRIPTION
**Pull request by YangFei1990**
1. documentation for biased_sampler.py
2. change the capital "GT" to "gt" for consistency 
3. some minor docs

### CODE REVIEW

File: MaskRCNN/model/biased_sampler.py [hunk 1]
```diff
@@ -9,27 +9,28 @@
 @under_name_scope()
 def sample_fast_rcnn_targets(boxes, gt_boxes, gt_labels, orig_gt_counts, batch_size):
     """
-    Sample some boxes from all proposals for training.
+    Sample boxes according to the predifined fg(foreground) boxes and bg(background) boxes ratio
```
Comment (karakusc) [inline MaskRCNN/model/biased_sampler.py]: Typo: "predefined"

File: MaskRCNN/model/biased_sampler.py [hunk 2]
```diff
@@ -9,27 +9,28 @@
...
     #fg(foreground) is guaranteed to be > 0, because ground truth boxes will be added as proposals.
     Args:
-        boxes: (#lvl x BS x K) x 5 region proposals. [batch_index, floatbox] aka Nx5
-        gt_boxes: BS x MaxGT x 4, floatbox
+        boxes: K x 5 region proposals. [batch_index, floatbox]
```
Comment (armandmcqueen) [inline MaskRCNN/model/biased_sampler.py]: Please clarify what a floatbox is here.

File: MaskRCNN/model/biased_sampler.py [hunk 3]
```diff
@@ -9,27 +9,28 @@
...
     Returns:
         sampled_boxes: tx5 floatbox, the rois
         sampled_labels: t int64 labels, in [0, #class). Positive means foreground.
-        fg_inds_wrt_gt: #fg indices, each in range [0, m-1].
-            It contains the matching GT of each foreground roi.
+        fg_inds_wrt_gt: #fg indices, each in range [0, m-1]. It contains the matching GT of each foreground roi.
     """
 
-    per_image_ious = pairwise_iou_batch(boxes, gt_boxes, orig_gt_counts, batch_size=batch_size) # list of len BS [N x M]
+    # Computes pairwise intersection-over-union between rpn boxes and gt boxes
```
Comment (armandmcqueen) [inline MaskRCNN/model/biased_sampler.py]: I think we can remove this line

File: MaskRCNN/model/biased_sampler.py [hunk 4]
```diff
@@ -9,27 +9,28 @@
...
+    # per_image_ious: list of len BS [N x M] -- N is Num_rpn_boxes and M is Num_gt_boxes, for one image
+    per_image_ious = pairwise_iou_batch(boxes, gt_boxes, orig_gt_counts, batch_size=batch_size)
 
     proposal_metrics_batch(per_image_ious)
 
 
-    ious = []
-    best_iou_inds = []
+    ious = [] # list of len BS [(N+M) x M]
```
Comment (armandmcqueen) [inline MaskRCNN/model/biased_sampler.py]: Confusing because it's an empty list here.

File: MaskRCNN/model/biased_sampler.py [hunk 5]
```diff
@@ -9,27 +9,28 @@
...
         sampled_boxes: tx5 floatbox, the rois
         sampled_labels: t int64 labels, in [0, #class). Positive means foreground.
-        fg_inds_wrt_gt: #fg indices, each in range [0, m-1].
-            It contains the matching GT of each foreground roi.
+        fg_inds_wrt_gt: #fg indices, each in range [0, m-1]. It contains the matching GT of each foreground roi.
     """
 
-    per_image_ious = pairwise_iou_batch(boxes, gt_boxes, orig_gt_counts, batch_size=batch_size) # list of len BS [N x M]
+    # Computes pairwise intersection-over-union between rpn boxes and gt boxes
+    # per_image_ious: list of len BS [N x M] -- N is Num_rpn_boxes and M is Num_gt_boxes, for one image
```
Comment (armandmcqueen) [inline MaskRCNN/model/biased_sampler.py:25]: Num_rpn_boxes is the same as num_rois, right?

File: MaskRCNN/utils/box_ops.py [hunk 6]
```diff
@@ -73,8 +73,8 @@ def pairwise_iou(boxlist1, boxlist2):
 def pairwise_iou_batch(proposal_boxes, gt_boxes, orig_gt_counts, batch_size):
     """Computes pairwise intersection-over-union between box collections.
     Args:
-      boxlist1: Nx5                             (batch_index, x1, y1, x2, t2)
-      boxlist2: BS x MaxNumGTs x 4
+      proposal_boxes: K x 5  (batch_index, x1, y1, x2, t2)
```
Comment (armandmcqueen) [inline MaskRCNN/utils/box_ops.py]: t2?

File: MaskRCNN/model/biased_sampler.py (hunk 5 above)
Comment (YangFei1990) [reply MaskRCNN/model/biased_sampler.py:25]: > Num_rpn_boxes is the same as num_rois, right?

This is the number of boxes before sampling. I would say the num_rois is the number of boxes after sampling.

File: MaskRCNN/model/biased_sampler.py [hunk 7]
```diff
@@ -134,7 +134,8 @@ def proposal_metrics_batch(per_image_ious):
     """
     Add summaries for RPN proposals.
     Args:
-        per_image_ios: list of len batch_size: nxm, #proposal x #gt
+         per_image_ios: pairwise intersection-over-union between rpn boxes and gt boxes
```
Comment (karakusc) [inline MaskRCNN/model/biased_sampler.py]: Typo: `per_image_ious`

File: MaskRCNN/model/generalized_rcnn.py [hunk 8]
```diff
@@ -293,8 +293,8 @@ def roi_heads(self, image, features, proposal_boxes, targets, inputs):
                 per_image_fg_labels = []
                 for i in range(cfg.TRAIN.BATCH_SIZE_PER_GPU):
 
-                    single_image_gt_count = prepadding_gt_counts[i] # 1-D Num_GT_boxes_current_image
-                    single_image_gt_masks = gt_masks[i, :single_image_gt_count, :, :] # Num_GT_boxes_current_image x H_gtmask x W_gtmask (maybe? might have length 1 dim at beginning)
+                    single_image_gt_count = prepadding_gt_counts[i] # 1-D Num_gt_boxes_current_image
+                    single_image_gt_masks = gt_masks[i, :single_image_gt_count, :, :] # Num_gt_boxes_current_image x H_gtmask x W_gtmask (maybe? might have length 1 dim at beginning)
```
Comment (karakusc) [inline MaskRCNN/model/generalized_rcnn.py]: Remove the comment in parenthesis. It will not have a dimension of 1 in the beginning.

**Pull request merged**